vsintegration/src/FSharp.Editor/LanguageService/FSharpCheckerExtensions.fs

```diff
@@ -38,7 +38,7 @@ type FSharpChecker with
         let tryGetFreshResultsWithTimeout() : Async<CheckResults> =
             async {
                 try
-                    let! worker = Async.StartChild(parseAndCheckFile, 2000)
+                    let! worker = Async.StartChild(parseAndCheckFile, 200)
```

Comment:
Make this a user option? I think the default could be anywhere from 500s to 2000s.  Accuracy is important....

Reply:
Agreed on this being a user option, and also a named parameter

Reply:
2 seconds is too long. Editing a large file which is type checked in > 2s means that _any_ completion takes > 2s, which feels very annoying because it's clear that we never get fresh check results in 2s for this file, but we still have to wait and wait and wait. Add to this the Roslyn artificial timeout and things are even worse. Even 1s is too long actually. About user setting, 99% of people are not aware of the Text Editor -> F# setting pages, so making it 2s by default + an option would make it just being 2s.

Reply:
Let's take this to the main discussion thread since it's the key part of the PR